README.md

```diff
@@ -32,130 +55,144 @@ FiloDB is a great fit for bulk analytical workloads, or streaming /  event data.
 
-Then, start running SQL queries!
+#### Spark data-source Example (spark-shell)
 
-### Detailed Ingestion Example
+You can follow along using the [Spark Notebook](http://github.com/andypetrella/spark-notebook) in doc/FiloDB.snb....  launch the notebook using `EXTRA_CLASSPATH=$FILO_JAR ADD_JARS=$FILO_JAR ./bin/spark-notebook &` where `FILO_JAR` is the path to `filodb-spark-assembly` jar.
 
-This example uses the checked in CSV file `GDELT-1979-1984-100000.csv` to walk through ingesting data.  To ingest the CSV, start spark-shell with the spark-csv package:
+Or you can start a spark-shell locally,
 
 ```bash
-bin/spark-shell --jars $REPOS/FiloDB/spark/target/scala-2.10/filodb-spark-assembly-0.1-SNAPSHOT.jar --packages com.databricks:spark-csv_2.10:1.2.0 --driver-memory 3G --executor-memory 3G
+bin/spark-shell --jars ../FiloDB/spark/target/scala-2.10/filodb-spark-assembly-0.1-SNAPSHOT.jar --packages com.databricks:spark-csv_2.10:1.2.0 --driver-memory 3G --executor-memory 3G
 ```
+Note that FiloDB uses an off-heap direct memory memtable, and you probably need to bump Spark's default direct memory size with a `-XX:MaxDirectMemorySize=1G` or some larger setting.
+
 
 Loading CSV file from Spark:
 
 ```scala
 scala> val csvDF = sqlContext.read.format("com.databricks.spark.csv").
            option("header", "true").option("inferSchema", "true").
-           load("GDELT-1979-1984-100000.csv")
+           load("../FiloDB/GDELT-1979-1984-100000.csv")
 ```
 
-Now, we have to decide on a sort and partitioning column.  The partitioning column decides how data is going to be distributed across the cluster, while the sort column acts as a primary key within each partition and decides how data will be sorted within each partition.
-
-If a partitioning column is not specified, FiloDB's Spark API will create a default one with the name "_partition", with a fixed value, which means everything will be thrown into one node, and is only suitable for small amounts of data.  The following command will add the default partition column, and create the dataset too:
-
+Creating a dataset from a Spark DataFrame,
 ```scala
 scala> import org.apache.spark.sql.SaveMode
 import org.apache.spark.sql.SaveMode
+
 scala> csvDF.write.format("filodb.spark").
-             option("dataset", "gdelt").option("sort_column", "GLOBALEVENTID").
+             option("dataset", "gdelt").
+             option("sort_column", "GLOBALEVENTID").
              mode(SaveMode.Overwrite).save()
 ```
 
-Note that FiloDB uses an off-heap direct memory memtable, and you probably need to bump Spark's default direct memory size with a `-XX:MaxDirectMemorySize=1G` or some larger setting.
+Or, specifying the `partition_column`,
 
-- Write data without partition column
+```scala
+scala> csvDF.write.format("filodb.spark").
+    option("dataset", "gdelt").
+    option("sort_column", "GLOBALEVENTID").
+    option("partition_column","ActionGeo_FullName").
```

Comment:
`ActionGeo_FullName` is not a good partitioning column, it generates too many partitions.   Instead use "Year".